### ISSUE TITLE
pin has invalid filename extension

### ISSUE THREAD

**Issue by Zahrun** [pacstall/pacstall#1151]
### Describe the bug

if the package name contains dots, apt will complain about the pin having invalid filename extension

### To reproduce

1. pacstall -I gir1.2-xapp-1.0-deb
2. sudo apt upgrade

### Expected behavior

pin filename has no dots

### Screenshots

```
~  sudo apt upgrade
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
N: Ignoring file 'gir1.2-xapp-1.0-deb-pin' in directory '/etc/apt/preferences.d/' as it has an invalid filename extension
```

### OS version

Tuxedo OS 3

### Pacstall version

5.2.1 (pacstall master)

**Comment by Elsie19** [comment]
@Zahrun can you run `pacstall -U Elsie19:fix-1151` and test it?

**Comment by Zahrun** [comment]
Yeah that fixes it.
Had to delete manually the previously wrong named filed but yes the fix-1151 version creates a pin which is not ignored and doesn’t trigger a warning

**Comment by oklopfer** [comment]
fixed in #1162
